### ISSUE TITLE
setTimeout(";") breaks JS Hint

### ISSUE THREAD

**Issue by electrovir** [AtomLinter/linter-jshint#288]
I get this error when linting:

```
Error: .atom/packages/linter-jshint/node_modules/jshint-json/json.js:4 console.log(JSON.stringify({ ^

TypeError: Converting circular structure to JSON at Object.stringify (native) at module.exports.reporter 
```

It shows up every it tries to lint on one of my files (which for me is triggered on file save) and completely breaks the linter from doing anything on my JS file.
I've discovered that one item breaks it:

``` javascript
setTimeout(";")
```

(I'm using this part of code because of the top voted answer here: http://stackoverflow.com/questions/3141064/how-to-stop-all-timeouts-and-intervals-using-javascript)

While it's probably indeed invalid or at least something to be warned about in Javascript, I think it shouldn't break the linter.

**Comment by Arcanemagus** [comment]
The error is being raised from the JSON validator within `jshint`, are you sure your `.jshintrc` file is valid JSON?

**Comment by electrovir** [comment]
Yes, it consists purely of this:

````json
{
  "esnext": true
}
````

This file is located in my user folder (the same folder that .atom is located in).  Is that the correct location?  It has worked as I intended so far from there.

**Comment by Arcanemagus** [comment]
`esnext` _used_ to be an option for the JSCS linter, but was removed in it's v3 version. As far as I know  JSHint only had the [`es3`](http://jshint.com/docs/options/#es3) and [`es5`](http://jshint.com/docs/options/#es5) options, which have been removed in favor of the [`esversion`](http://jshint.com/docs/options/#esversion) option. Try changing it to look like this:

```json
{
  "esversion": 6
}
```

**Comment by electrovir** [comment]
K, I just changed my ````.jshintrc```` file as you advised, but the same error still occurs.

**Comment by Arcanemagus** [comment]
Looks like this is a duplicate of https://github.com/sindresorhus/jshint-json/issues/1, seeing if I can trace it down...

**Comment by Arcanemagus** [comment]
Looks like this is being ran as an external process, so debugging it within Atom isn't as simple as I thought.

**Comment by Arcanemagus** [comment]
Marking this as closed since it has nothing to do with this package. Please follow the issue on `jshint-json` linked above for further developments.
